src/test/serfloat_tests.cpp

```diff
@@ -47,52 +48,35 @@ BOOST_AUTO_TEST_CASE(double_serfloat_tests) {
-        TestDouble(std::numeric_limits<double>::min());
-        TestDouble(-std::numeric_limits<double>::min());
-        TestDouble(std::numeric_limits<double>::max());
-        TestDouble(-std::numeric_limits<double>::max());
-        TestDouble(std::numeric_limits<double>::lowest());
-        TestDouble(-std::numeric_limits<double>::lowest());
-        TestDouble(std::numeric_limits<double>::quiet_NaN());
-        TestDouble(-std::numeric_limits<double>::quiet_NaN());
-        TestDouble(std::numeric_limits<double>::signaling_NaN());
-        TestDouble(-std::numeric_limits<double>::signaling_NaN());
-        TestDouble(std::numeric_limits<double>::denorm_min());
-        TestDouble(-std::numeric_limits<double>::denorm_min());
-        // On IEC559-compatible systems, construct doubles to test from the encoding.
-        for (int j = 0; j < 1000; ++j) {
-            // Iterate over 9 specific bits exhaustively; the others are chosen randomly.
-            // These specific bits are the sign bit, and the 2 top and bottom bits of
-            // exponent and mantissa in the IEEE754 binary64 format.
-            for (int x = 0; x < 512; ++x) {
-                uint64_t v = InsecureRandBits(64);
-                v &= ~(uint64_t{1} << 0);
-                if (x & 1) v |= (uint64_t{1} << 0);
-                v &= ~(uint64_t{1} << 1);
-                if (x & 2) v |= (uint64_t{1} << 1);
-                v &= ~(uint64_t{1} << 50);
-                if (x & 4) v |= (uint64_t{1} << 50);
-                v &= ~(uint64_t{1} << 51);
-                if (x & 8) v |= (uint64_t{1} << 51);
-                v &= ~(uint64_t{1} << 52);
-                if (x & 16) v |= (uint64_t{1} << 52);
-                v &= ~(uint64_t{1} << 53);
-                if (x & 32) v |= (uint64_t{1} << 53);
-                v &= ~(uint64_t{1} << 61);
-                if (x & 64) v |= (uint64_t{1} << 61);
-                v &= ~(uint64_t{1} << 62);
-                if (x & 128) v |= (uint64_t{1} << 62);
-                v &= ~(uint64_t{1} << 63);
-                if (x & 256) v |= (uint64_t{1} << 63);
-                double f;
-                memcpy(&f, &v, 8);
-                TestDouble(f);
-            }
+    // Test extreme values
+    TestDouble(std::numeric_limits<double>::min());
+    TestDouble(-std::numeric_limits<double>::min());
+    TestDouble(std::numeric_limits<double>::max());
+    TestDouble(-std::numeric_limits<double>::max());
+    TestDouble(std::numeric_limits<double>::lowest());
+    TestDouble(-std::numeric_limits<double>::lowest());
+    TestDouble(std::numeric_limits<double>::quiet_NaN());
+    TestDouble(-std::numeric_limits<double>::quiet_NaN());
+    TestDouble(std::numeric_limits<double>::signaling_NaN());
+    TestDouble(-std::numeric_limits<double>::signaling_NaN());
+    TestDouble(std::numeric_limits<double>::denorm_min());
+    TestDouble(-std::numeric_limits<double>::denorm_min());
+    // Construct doubles to test from the encoding.
```

Comment:
IIUC pseudocode:
Use values of `x` to codify all possibilities of 9 bits, with 1000 iters for each setting. So x=0..(1000*2^9)-1
for `x_pos`, `v_pos` in enumerate([0, 1, 50, 51, 52, 53, 61, 62, 63]):
&nbsp;&nbsp;&nbsp;&nbsp;`v` starts as 64 random bits
&nbsp;&nbsp;&nbsp;&nbsp;reset `v`'s `v_pos`th bit to what `x` 's `x_pos` bit is
&nbsp;&nbsp;&nbsp;&nbsp;turn `v` into double `f` and `TestDouble(f)`